File: Java/LifecycleAware/app/src/main/java/com/example/lifecycleawarecomponents/SecondActivity.java
package com.example.lifecycleawarecomponents;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

public class SecondActivity extends AppCompatActivity {
    private static final String TAG = "MYTAG";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_second);
        getLifecycle().addObserver(new DemoAppComponent("SecondActivity"));

        Log.i(TAG,"***************    SecondActivity onCreate() ");
    }

    @Override
    protected void onResume() {
        super.onResume();
        Log.i(TAG,"***************    SecondActivity onResume() ");
    }
}

File: Kotlin/DataBindingDemo/app/src/main/java/com/example/databindingdemo/SecondActivity.kt
package com.example.databindingdemo

import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.example.databindingdemo.databinding.ActivitySecondBinding

class SecondActivity : AppCompatActivity() {
    private lateinit var binding: ActivitySecondBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_second)

        binding.student = getStudent()
    }

    private fun getStudent(): Student {
        return Student(1, "Edgar", "test@gmail.com")
    }
}
